File: 189-rotate-array/rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    const length = nums.length;
    const breakpoint = k % length;
    if (breakpoint === 0) return;

    const arr1 = nums.slice(-breakpoint);
    const arr2 = nums.slice(0, length - breakpoint);
    nums.splice(0, length, ...arr1, ...arr2);
};

File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        breakpoint = k % length
        if breakpoint == 0:
            return

        arr1 = nums[length - breakpoint:]
        arr2 = nums[:length - breakpoint]
        nums[0:0+len(arr1)] = arr1
        nums[breakpoint: breakpoint + (len(arr2))] = arr2

        
        